src/util/base58.rs

```diff
@@ -134,14 +177,12 @@ pub fn from_check(data: &str) -> Result<Vec<u8>, Error> {
     Ok(ret)
 }
 
-fn encode_iter_utf8<I>(data: I) -> Vec<u8>
+fn encode_iter_utf8<I, W>(data: I, writer: &mut W) -> Result<(), fmt::Error>
```

Comment:
It is obvious to me that the function should be renamed since it'f no longer returning UTF-8 bytes but rather writing unicode `char`s to a writer. What would you think is a good name instead?

Reply:
sounds good